rust/altrios-core/src/train/speed_limit_train_sim.rs

```diff
@@ -288,31 +290,44 @@ impl SpeedLimitTrainSim {
     }
 
     pub fn solve_step(&mut self) -> anyhow::Result<()> {
+
+        //set catenary power limit.  This was the same as set_speed_train_sim.  I am assuming it is 0 right now.
         self.loco_con
             .set_cat_power_limit(&self.path_tpc, self.state.offset);
+        //set aux power for the consist.  Very similar to set speed train sim.  boolean argument is whether engine is on.  Not a huge fane of this argument.  I think this should be baked into a loco control system in locomotive.rs
         self.loco_con.set_pwr_aux(Some(true))?;
+        //set the maximum power out based on dt.  This is different than set speed train sim because that one calls the speed trace dt.
+        //why are we not passing in aux power here and self.state?  Am I missing something obviouls here?
         self.loco_con.set_cur_pwr_max_out(None, self.state.dt)?;
+        //calculate new resistance. same as set_speed_train_sim.
         self.train_res
             .update_res(&mut self.state, &self.path_tpc, &Dir::Fwd)?;
+        //solve the required power.  No argument is passed here unlike set_speed_train sim..
+        //I'm about to figure out why maybe......
         self.solve_required_pwr()?;
         #[cfg(feature = "logging")]
         log::debug!(
             "{}\ntime step: {}",
             format_dbg!(),
             self.state.time.get::<si::second>().format_eng(Some(9))
         );
+
+        //this is similar to set_speed train sim, but speedtrace.dt is passed rather than state.dt
+        //could we get away with passing state.dt instead of speed_trace.dt in set_speed_train_sim?
+        //could this cause a lag of 1 dt in speed which would be a small but consistent offset in the calcs?
```

Comment:
@SWRIganderson , hopefully https://github.com/NREL/altrios/pull/63/files#r1755701856 provides an explanation of why we would not want to do this.